instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/NetAttributesExtractor.java

```diff
@@ -26,10 +31,6 @@ protected final void onStart(AttributesBuilder attributes, REQUEST request) {
   @Override
   protected final void onEnd(AttributesBuilder attributes, REQUEST request, RESPONSE response) {
     set(attributes, SemanticAttributes.NET_PEER_IP, peerIp(request, response));
-
-    // TODO(anuraaga): Clients don't have peer information available during the request usually.
```

Comment:
But related to my above comment, actually this isn't quite solved since it means instrumentation defining separate ServerNetAttributesExtractor and ClientNetAttributesExtractor, which our pattern doesn't really support since it makes instrumentation more complex.  The approach with `@Nullable RESPONSE response` could solve it though.

Reply:
I've refactored the `NetAttributesExtractor` class so that it has a `@Nullable` response in its attribute methods.